toolkit/tools/srpmdownloader/srpmdownloader.go

```diff
@@ -0,0 +1,232 @@
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/exe"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/logger"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/network"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/retry"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/rpm"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/safechroot"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/srpm"
+
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+const (
+	defaultBuildDir    = "./build/SRPMS"
+	defaultWorkerCount = "10"
+)
+
+type sourceRetrievalConfiguration struct {
+	sourceURLs []string
+	caCerts    *x509.CertPool
+	tlsCerts   []tls.Certificate
+}
+
+var (
+	app = kingpin.New("srpmpacker", "A tool to package a SRPM.")
+
+	specsDir = exe.InputDirFlag(app, "Path to the SPEC directory to create SRPMs from.")
+	outDir   = exe.OutputDirFlag(app, "Directory to place the output SRPM.")
+	logFile  = exe.LogFileFlag(app)
+	logLevel = exe.LogLevelFlag(app)
+
+	buildDir = app.Flag("build-dir", "Directory to store temporary files while building.").Default(defaultBuildDir).String()
+	distTag  = app.Flag("dist-tag", "The distribution tag SRPMs will be built with.").Required().String()
+	runCheck = app.Flag("run-check", "Whether or not to run the spec file's check section during package build.").Bool()
+
+	workers = app.Flag("workers", "Number of concurrent goroutines to parse with.").Default(defaultWorkerCount).Int()
+
+	// Use String() and not ExistingFile() as the Makefile may pass an empty string if the user did not specify any of these options
+	srpmUrlList   = app.Flag("srpm-url-list", "urls for SRPM repo.").String()
+	srpmListFile  = app.Flag("srpm-list", "Path to a list of SPECs to pack. If empty will pack all SPECs.").ExistingFile()
+	caCertFile    = app.Flag("ca-cert", "Root certificate authority to use when downloading files.").String()
+	tlsClientCert = app.Flag("tls-cert", "TLS client certificate to use when downloading files.").String()
+	tlsClientKey  = app.Flag("tls-key", "TLS client key to use when downloading files.").String()
+
+	workerTar = app.Flag("worker-tar", "Full path to worker_chroot.tar.gz. If this argument is empty, SRPMs will be packed in the host environment.").ExistingFile()
+)
+
+func main() {
+	app.Version(exe.ToolkitVersion)
+	kingpin.MustParse(app.Parse(os.Args[1:]))
+	logger.InitBestEffort(*logFile, *logLevel)
+
+	if *workers <= 0 {
+		logger.Log.Fatalf("Value in --workers must be greater than zero. Found %d", *workers)
+	}
+
+	// Create a template configuration that all packed SRPM will be based on.
+	var templateSrcConfig sourceRetrievalConfiguration
+	// Setup remote source configuration
+	var err error
+	templateSrcConfig.caCerts, err = x509.SystemCertPool()
+	logger.PanicOnError(err, "Received error calling x509.SystemCertPool(). Error: %v", err)
+	if *caCertFile != "" {
+		newCACert, err := ioutil.ReadFile(*caCertFile)
+		if err != nil {
```

Comment:
I think we can replace the `err != nil` check with `PanicOnError()`. Similar comment for the TLS cert check below.